easybuild/easyconfigs/y/Yambo/Yambo-5.2.dev-20230512-intel-2021b.eb

```diff
@@ -0,0 +1,85 @@
+easyblock = 'MakeCp'
+
+name = 'Yambo'
+version = '5.2.dev-20230512'
+local_yambo_libs_ver = '1.0'
+local_commit = 'dfa2752'
+
+homepage = 'http://www.yambo-code.org'
+description = """Yambo is a FORTRAN/C code for Many-Body calculations in solid state and molecular physics.
+ Yambo relies on the Kohn-Sham wavefunctions generated by two DFT public codes: abinit, and PWscf."""
+
+toolchain = {'name': 'intel', 'version': '2021b'}
+toolchainopts = {'usempi': True, 'openmp': True}
+
+local_yambo_ext_cmd = 'mkdir yambo-%(version)s && '
+local_yambo_ext_cmd += 'tar --strip-components=1 -xzf %s -C yambo-%(version)s'
+local_yambo_lib_ext_cmd = 'mkdir yambo-%(version)s/lib/yambo/ && '
+local_yambo_lib_ext_cmd += 'tar --strip-components=1 -xzf %s -C yambo-%(version)s/lib/yambo/'
+
+source_urls = ['https://github.com/yambo-code/yambo/archive/']
+sources = [
+    {
+        'download_filename': '%s.tar.gz' % local_commit,
+        'filename': 'yambo-%(version)s.tar.gz',
+        'extract_cmd': local_yambo_ext_cmd,
+    },  # Yambo
+    {
+        'source_urls': ['https://github.com/yambo-code/yambo-libraries/archive/'],
+        'download_filename': '%s.tar.gz' % local_yambo_libs_ver,
+        'filename': 'yambo-libraries-%s.tar.gz' % local_yambo_libs_ver,
+        'extract_cmd': local_yambo_lib_ext_cmd,
+    },
+    {
+        'source_urls': ['https://github.com/yambo-code/yambo/files/962173/'],
+        'filename': 'iotk-y1.2.2.tar.gz',
+        'extract_cmd': 'cp %s yambo-%(version)s/lib/archive/',
+    },
+]
+checksums = [
+    ('4a993b86543fec7c5fa79838ab699a9f4432a215f63eca537c0819335ab9981c',
+    'f25d86101cc9cead9f64f05181aab1f06eac8528fea5ba5c586b2b9512cb8492',
+    'c0a4eb19f3e885d83d7afa52eb90658fba7cb1cb6e66049866a98dcc980de543')
```

Comment:
@ItIsI-Orient This syntax is definitely incorrect. You have a list with a single element, which is a tuple with 3 values.

Although that's supported, it doesn't specify what you probably intend to: if gives 3 different valid checksums for the 1st source file (and hence there are no checksums at all provided for the 2nd and 3rd source).

This should work (a list with 3 elements):

```python
checksums = [
    '4a993b86543fec7c5fa79838ab699a9f4432a215f63eca537c0819335ab9981c',
    'f25d86101cc9cead9f64f05181aab1f06eac8528fea5ba5c586b2b9512cb8492',
    'c0a4eb19f3e885d83d7afa52eb90658fba7cb1cb6e66049866a98dcc980de543',
]
```

This should also work:

```python
checksums = [
    {'yambo-5.2.dev-20230512.tar.gz': '4a993b86543fec7c5fa79838ab699a9f4432a215f63eca537c0819335ab9981c'},
    {'yambo-libraries-1.0.tar.gz': 'f25d86101cc9cead9f64f05181aab1f06eac8528fea5ba5c586b2b9512cb8492'},
    {'iotk-y1.2.2.tar.gz': 'c0a4eb19f3e885d83d7afa52eb90658fba7cb1cb6e66049866a98dcc980de543'},
]
```

Note that the keys must match with the filenames of the sources as they get stored in the source path, or you'll hit a confusing error like:

```
Invalid checksum spec 'None', should be a string (MD5) or 2-tuple (type, value).
```

Reply:
The latter is actually a bug, see also https://github.com/easybuilders/easybuild-framework/pull/4150